tests/flow_e2e_tests.py

```diff
@@ -59,3 +61,14 @@ def test_no_flyby_trigger_flow(self):
         flow_message = self.bot.pop_message()
         self.assertIn('You are in the flow w1', flow_message)
         self.assertEqual(len(self.bot.flow_executor.in_flight), 1)
+
+    def test_flow_only(self):
+        self.assertCommand('!a', 'a')  # non flow_only should respond.
+        self.bot.push_message('!d')
+        self.assertRaises(Empty, self.bot.pop_message)
```

Comment:
I suggest using `timeout=1` on the `pop_message` call, otherwise it will wait for a 5-second timeout. This command executes so fast when it actually triggers, a 1 second timeout should be long enough to be certain it didn't trigger.